akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -361,15 +361,22 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
   }
 
   def PendingBufferWrite(commander: ActorRef, data: ByteString, ack: Event, tail: WriteCommand): PendingBufferWrite = {
-    val buffer = bufferPool.acquire()
-    try {
-      val copied = data.copyToBuffer(buffer)
-      buffer.flip()
-      new PendingBufferWrite(commander, data.drop(copied), ack, buffer, tail)
-    } catch {
-      case NonFatal(e) ⇒
-        bufferPool.release(buffer)
-        throw e
+    if (data.headByteString.canWrapAsByteBuffer) {
```

Comment:
this may have the downside of committing tiny writes (as sparse ByteString parts might be small) to the channel. if that is not ideal, this check can easily account for size and make dense buffers when needed (e.g. things below the standard MTU)